policy/evaluator.go

```diff
@@ -297,22 +298,36 @@ func (eval *evaluator) Eval(vars interface{}) ([]ref.Val, error) {
 	}
 	decisions := []ref.Val{}
 	terms := eval.activation.terms
-	activation := &evaluatorActivation{
-		input:     varActivation,
-		terms:     terms,
-		memoTerms: make(map[string]ref.Val),
+	activation := evalActivationPool.Get().(*evaluatorActivation)
+	activation.input = varActivation
+	activation.terms = terms
+	for k := range activation.memoTerms {
+		delete(activation.memoTerms, k)
 	}
+	errors := []error{}
 	for _, rule := range eval.rules {
 		matches, _, err := rule.match.program.Eval(activation)
 		if err != nil {
-			return nil, err
+			errors = append(errors, err)
+			continue
 		}
 		if matches != types.True {
 			continue
 		}
 		output, _, err := rule.output.program.Eval(activation)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
 		decisions = append(decisions, output)
 	}
+	evalActivationPool.Put(activation)
+	if len(decisions) != 0 {
```

Comment:
Is this intending to mask errors if there's at least one decision?  I think this might need a comment explaining the intent.

Reply:
sounds good, until then, maybe log a warning if we see errors?

Reply:
Done.